data-processing/common/commands/fetch_s2_data.py

```diff
@@ -99,13 +102,28 @@ def process(self, item: ArxivId) -> Iterator[S2Metadata]:
                 data = resp.json()
                 references = []
                 for reference_data in data["references"]:
+
+                    if reference_data["paperId"] is None:
+                        # can be legit references not in our corpus, or bad references from bogus extractions
+                        continue
```

Comment:
* As mentioned in this [comment](https://github.com/allenai/scholarphi/pull/377/files#r1097999664)...
  * New public API endpoint gives us _all_ the references, even for those that failed Citation Matching (and that's why both paperID/SHA and corpus ID are blank).  
  * Old API already ignore those, so here we're making sure the behavior is not changed.
* PEOPLE MIGHT ASK... ok so the new API does provide legit references which just happen to not be in our corpus, wouldn't it be good to include those anyway?
  * No... currently this group of id-less failed citation matches ALSO includes a lot of garbage from bogus extractions.  We can't tell (without human intervention) which is which.  So right now it's just safer to exclude this noise from getting passed into the Reader.  
  * Besides, the inline mentions should (irrelev to this ticket) already show Citation Cards with general error msgs.  
  * Even if we do pass in these id-less papers, it's not like there's a valid PDP to send users to anyway.  YAGNI.

Reply:
Ok, so I reran the testing (as seen in IDEA 2C from this PR's main comment), and the resulting `citations.csv` is still the same, as expected.

The only difference is that I can see from my temporary print statements (green) that those 3 unwanted no-id references from the NEW public API endpoint have been excluded, yay.
![Screen Shot 2023-02-06 at 5 33 36 PM](https://user-images.githubusercontent.com/47910008/217125552-a7a84de5-6d41-457d-b464-847d218fe858.png)